tests/sentry/workflow_engine/migration_helpers/test_migrate_alert_rule.py

```diff
@@ -152,254 +151,154 @@ def assert_alert_rule_trigger_action_migrated(alert_rule_trigger_action, action_
+        )
+        (
+            workflow,
+            detector,
+            detector_workflow,
+            _,  # the workflow trigger group for a migrated metric alert rule is None
+        ) = self.create_detector_and_workflow(workflow_triggers=None)
+        detector.update(workflow_condition_group=detector_data_condition_group)
+        detector_state = self.create_detector_state(
+            detector=detector,
+            active=False,
+            state=DetectorPriorityLevel.OK,
+        )
+        data_source_detector = self.create_data_source_detector(data_source, detector)
+        (
+            alert_rule_detector,
+            alert_rule_workflow,
+        ) = self.create_metric_alert_lookup_tables(metric_alert, detector, workflow)
+
+        return (
+            data_source,
+            detector_data_condition_group,
+            workflow,
+            detector,
+            detector_state,
+            alert_rule_detector,
+            alert_rule_workflow,
+            detector_workflow,
+            data_source_detector,
         )
 
-        assert resolve_detector_trigger.type == Condition.GREATER_OR_EQUAL
-        assert resolve_detector_trigger.comparison == critical_trigger.alert_threshold
-        assert resolve_detector_trigger.condition_result == DetectorPriorityLevel.OK
-        assert resolve_detector_trigger.condition_group == detector.workflow_condition_group
-
-    def test_create_metric_alert_trigger_action(self):
-        """
-        Test that when we call the helper methods we create all the ACI models correctly for alert rule trigger actions
-        """
-        migrate_alert_rule(self.metric_alert, self.rpc_user)
 
-        migrate_metric_data_conditions(self.alert_rule_trigger_warning)
-        migrate_metric_data_conditions(self.alert_rule_trigger_critical)
+class DualDeleteAlertRuleTest(BaseMetricAlertMigrationTest):
+    def setUp(self):
+        self.metric_alert = self.create_alert_rule()
+        (
+            self.data_source,
+            self.detector_data_condition_group,
+            self.workflow,
+            self.detector,
+            self.detector_state,
+            self.alert_rule_detector,
+            self.alert_rule_workflow,
+            self.detector_workflow,
+            self.data_source_detector,
+        ) = self.set_up_migrated_metric_alert_objects(self.metric_alert)
+
+    def test_dual_delete_metric_alert(self):
+        dual_delete_migrated_alert_rule(self.metric_alert)
+        with self.tasks():
+            run_scheduled_deletions()
 
-        migrate_metric_action(self.alert_rule_trigger_action_email)
-        migrate_metric_action(self.alert_rule_trigger_action_integration)
-        migrate_metric_action(self.alert_rule_trigger_action_sentry_app)
+        # check workflow-related tables
+        assert not Workflow.objects.filter(id=self.workflow.id).exists()
+        assert not AlertRuleWorkflow.objects.filter(id=self.alert_rule_workflow.id).exists()
```

Comment:
This seems like it might be a good separate test rather than 1 assertion chunk. The reason for that is if we fail to remove a workflow, we won't know if the detector stuff is deleting correctly or not because this test will stop execution at the first failed assertion.